src/Sulu/Component/Content/Document/SynchronizationManager.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document;
+
+use Sulu\Bundle\ContentBundle\Document\RouteDocument;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentInspector;use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentManagerRegistry;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\PropertyEncoder;
+use Sulu\Component\Content\Document\Behavior\ResourceSegmentBehavior;
+use Sulu\Component\Content\Document\Behavior\SynchronizeBehavior;
+use Sulu\Component\Content\Document\Syncronization\DocumentRegistrator;
+use Sulu\Component\DocumentManager\Behavior\Mapping\LocaleBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerRegistryInterface;
+
+/**
+ * The synchronization manager handles the synchronization of documents
+ * from the DEFAULT document manger to the PUBLISH document manager.
+ *
+ * NOTE: In the future multiple document managers may be supported.
+ *
+ * NOTE: Much of the logic in this class is about ensuring that corresponding
+ *       nodes exist in the PDM - this could be simplified (and made much more
+ *       efficient) if we were to use Jackalope observation to implement Event
+ *       listeners at the PHPCR level, however this feature is currently not
+ *       supported.
+ *
+ *       See: https://github.com/jackalope/jackalope/pull/241
+ *
+ * TODO: Explcitly inject publish and default document managers?
```

Comment:
Should we create an explicit service alias for the publish document manager? and inject it here instead of the `liveWorkspacename` and the `ManagerRegistry` ?